src/main/scala/scala/util/parsing/combinator/RegexParsers.scala

```diff
@@ -137,8 +137,19 @@ trait RegexParsers extends Parsers {
     }
   }
 
+  /**
+   * A parser generator delimiting whole phrases (i.e. programs).
+   *
+   * `phrase(p)` succeeds if `p` succeeds and no input is left over after `p`.
+   *
+   * @param p the parser that must consume all input for the resulting parser
+   *          to succeed.
+   *
+   * @return  a parser that has the same result as `p`, but that only succeeds
+   *          if `p` consumed all the input.
+   */
   override def phrase[T](p: Parser[T]): Parser[T] =
-    super.phrase(p <~ opt("""\z""".r))
```

Comment:
@gourlaysama This line was added before to fix this [issue](https://issues.scala-lang.org/browse/SI-1178) to consume the trailing whitespace at the end of a `phrase` parser. But that produced a new issue that I have mentioned. I think the proper fix is to use regex that match nothing.